source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -1279,12 +1301,15 @@ implementation::
                     if isNotMaster(error):
                         check failing server
             elif isNetworkError(error) or (not error.completedHandshake and (isNetworkTimeout(error) or isAuthError(error))):
-                # Mark the server Unknown
-                unknown = new ServerDescription(type=Unknown, error=error)
-                onServerDescriptionChanged(unknown, connection pool for server)
-                clear connection pool for server
-                # Cancel inprogress check
-                cancel monitor check
+                if type != LoadBalanced
+                  # Mark the server Unknown
+                  unknown = new ServerDescription(type=Unknown, error=error)
+                  onServerDescriptionChanged(unknown, connection pool for server)
+                  clear connection pool for server
+                  # Cancel inprogress check
+                  cancel monitor check
+                else
+                  clear connection pool for serviceId
```

Comment:
If `not error.completedHandshake`, this should only happen if `serviceId` is set as that indicates that the initial `hello` succeeded.

Reply:
Added check for serverId existence.